src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -22,12 +32,16 @@
           (log/warn (format "%s %s"
                             (i18n/trs "The ''master.allow-header-cert-info'' setting is deprecated and will be ignored in favor of the ''authorization.allow-header-cert-info'' setting because the ''jruby-puppet.use-legacy-auth-conf'' setting is ''false''.")
                             (i18n/trs "Remove the ''master.allow-header-cert-info'' setting.")))))
-      (assoc context :request-handler
-                     (request-handler-core/build-request-handler
-                      jruby-service
-                      (request-handler-core/config->request-handler-settings
-                       config)
-                      current-code-id))))
+      ;; TODO: Throw error when max-active-requests > 0, but metrics-service is nil
```

Comment:
Hmm, actually I guess the `:JRubyMetricsService` essentially was already required via the `JRubyPuppetService` for the setup work we did around the http-client metrics in [this PR](https://github.com/puppetlabs/puppetserver/pull/1322).  I think it's probably fine to just go with making `:JRubyMetricsService` a required dependency, allowing us to do away with the check for `nil` handling as @jpinsonault had suggested.  I'm also fine with just trying to mine the active request count out of the metrics service info.